ccmbioinfo/lfs
# CARDS-1886: Replace Torch import with Clarity import

Also includes CARDS-2208: Clarity import: NOT IN rules for configured mappers and filters don't work as expected

To test proms:

1. build with `mvn clean install -Pdocker`
2. in `compose-cluster/mssql` run `python3 generate_test_proms_sql.py -n 100 proms_sample.sql`
3. in `compose-cluster` run `python3 generate_compose_yaml.py --mssql --cards_project cards4proms --oak_filesystem --dev_docker_image --smtps --smtps_test_container --smtps_test_mail_path ~/cardsmail --composum --adminer --clarity`
4. in `compose-cluster` start cards with `docker-compose up`
5. at `http://localhost:1435/?mssql=mssql&username=sa&db=master&ns=path&import=` import the generated `proms_sample.sql` file
6. open `http://localhost:8080/Subjects.importClarity`
7. check that the patients have been imported correctly
8. stop docker-compose and run `docker-compose rm` and `./cleanup.sh`
9. repeat steps 3-5
10. at `http://localhost:8080/system/console/configMgr` search for `NightlyClarityImport` and change the configuration to have the import schedule at about 2 minutes from now (the format is `seconds minute hour ...`, so change it to something like `0 45 14 * * ? *` to run at `14:45`)
11. wait for 2 minutes, then check that the forms have been imported

Regression testing for prems:

1. build with `mvn clean install -Pdocker`
2. in `compose-cluster/mssql` run `python3 ./generate_test_YE_sql.py -n 100 --basedate 2023-04-24 --time_spread_seconds 0 prems_sample.sql` (use 7 days ago as the date)
3. in `compose-cluster` run `python3 generate_compose_yaml.py --mssql --cards_project cards4prems --oak_filesystem --dev_docker_image --smtps --smtps_test_container --smtps_test_mail_path ~/cardsmail --composum --adminer --clarity`
4. in `compose-cluster` start cards with `docker-compose up`
5. at `http://localhost:1435/?mssql=mssql&username=sa&db=master&ns=path&import=` import the generated `prems_sample.sql` file
6. open `http://localhost:8080/Subjects.importClarity?dayToQuery=-7`
7. check that the patients have been imported correctly
8. stop docker-compose and run `docker-compose rm` and `./cleanup.sh`
9. repeat steps 3-5
10. at `http://localhost:8080/system/console/configMgr` search for `NightlyClarityImport` and change the configuration to have the import schedule at about 2 minutes from now (the format is `seconds minute hour ...`, so change it to something like `0 45 14 * * ? *` to run at `14:45`)
11. wait for 2 minutes, then check that the forms have been imported


To do:
- [x] Rename the clinic ID to `TG-PMCC CARDIAC CLINICS DIAGNOSTIC`
- [x] Copy the clinic name to the Visit Information/Location answer
- [x] Special handling for Canceled events: if previously imported, update the existing Visit Information/Status answer 
- [x] Discard visits without DOB or both `MRN` and `OHIP_NBR`
- [x] Add a processor that automatically maps between `ENCOUNTER_CLINIC` and Visit Information/Clinic
